### ISSUE TITLE
Utilize new golang 1.8 features within the codebase

### ISSUE THREAD

**Issue by Roasbeef** [lightningnetwork/lnd#276]
# Issue 

With golang 1.9 out, we'll be sliding over by a version, making the minimal version supported by `lnd` to be golang 1.8 and 1.9. As a result, we can now safely use new language features or package related changes that are now relevant. Off the top, I can think of two such instances, though it's possible that there are many more. 

# Steps to Completion 

- [x] We can now use the [`sort.Slice`](https://golang.org/pkg/sort/#Slice) function rather than manually re-creating the `sort.Sort` interface for each instance where we want to be able to sort a slice. A quick grep shows the instances within the codebase where this function can be used instead:
    ```
    ⛰ grep -R -n "sort.Sort" --exclude-dir=vendor .
    ./lnwallet/channel.go:2362:	sort.Sort(sortedSigs)
    ./routing/heap_test.go:38:	sort.Sort(&distanceHeap{sortedEntries})
    ./routing/router.go:877:	sort.Sort(validRoutes)
   ```

- [x] The second instance I've identified is that [binary.Read/Write](https://golang.org/pkg/encoding/binary/#Read) now accepts `bool` arguments. So within `channeldb`, we can use this [rather than creating our own version](https://github.com/lightningnetwork/lnd/blob/master/channeldb/channel.go#L2351). There are a few other instances that don't use that function, which can be found by grepping for `[1]byte` within the codebase.

**Comment by flaurida** [comment]
I am going to tackle the first part of this (first checkbox) :)

**Comment by flaurida** [comment]
Still getting my feet wet but here's my code for replacing the first sort.Sort
https://github.com/lightningnetwork/lnd/pull/315

**Comment by Roasbeef** [comment]
This was fixed by #315, closing.
